chess-AI/ai_io.py

```diff
@@ -166,29 +235,59 @@ def init_params(nnet, device):
         raise ValueError("This program must be run with the `train.sh` script. See the script "
                          "for usage instructions.")
 
-    # Load in a model
-    if model_path is not None:
-        print()
-        print(model_path)
-        print()
-        if not os.path.exists(os.path.dirname(model_path)):
-            os.makedirs(os.path.dirname(model_path))
-        load_model(nnet, model_path, num_saved_models)
+    # wait until training has begun and catch any param errors
+    if start_train:
+        if ds_saving["data_dir"] is None:
+            raise ValueError("Directory for dataset must always be specified")
+        else:
+            assert os.path.exists(os.path.dirname(ds_saving["data_dir"])), \
+                "Dataset load directory does not exist"
+
+        if m_saving["model_dir"] is None:
+            raise ValueError("Directory for model must always be specified.")
+        else:
+            assert os.path.exists(os.path.dirname(m_saving["model_dir"])), \
+                "Model load directory does not exist"
+
+        if ds_saving["local_load"] or ds_saving["figshare_load"]:
+            if not ds_saving["file_name"]:
+                raise ValueError("A file name must be specified to load a dataset")
+            if make_dataset:
+                raise ValueError("Dataset cannot be created and loaded")
+        elif not make_dataset:
+            raise ValueError("If a dataset is not being created then local_load or \
+                figshare_load must be specified")
+
+        if m_saving["local_load"] or m_saving["figshare_load"]:
+            if not m_saving["file_name"]:
+                raise ValueError("A file name must be specified to load a model")
+
+        if ds_saving["local_load"] and ds_saving["figshare_load"]:
+            raise ValueError("Dataset cannot be loaded from figshare and locally")
+
+        if m_saving["local_load"] and m_saving["figshare_load"]:
+            raise ValueError("Model cannot be loaded from figshare and locally")
+
+        # if no errors are raised then model can be loaded
+        if m_saving["local_load"] or m_saving["figshare_load"]:
+            load_model(nnet, m_saving, args.dashboard)
```

Comment:
To save an if statement, you can put this after checking the file name on line 262/263